### DISCUSSION TITLE
why trace id bind with thread and it won't reset

### DISCUSSION THREAD

**Discussion by liu-junchi** [apache/skywalking#7871] [Q&A]
Java agent, version: 8.1.0

This trace link has been maintained for many days and has not ended
![image](https://user-images.githubusercontent.com/12181207/136513094-892bf777-5687-4067-a2e7-8d913c27c544.png)

Then I find they have same thread, like 'DubboServerHandler-10.253.130.106:30013-thread-200', if this thread process request, the trace id is always same.

Why this happen, hope for help, thanks.

**Comment by wu-sheng** [comment]
It means you have faced some tracing context leak. The reason is uncertain, maybe some plugins don't match your cases.

**Reply by liu-junchi** [reply]
Could it be caused by asynchronous propagation?

**Reply by wu-sheng** [reply]
I am afraid mostly not. Async scenario usually causes trace breaking, rather than leaking.

**Reply by liu-junchi** [reply]
Not every time this request link will cause this result, it looks like it’s random, and it’s hard to find the reason.  :(

**Reply by wu-sheng** [reply]
It could be random or thread reuse model. It is hard to determine from this general description.

**Reply by liu-junchi** [reply]
In most cases, random events may be related to multi-threaded scenarios, so I just guessed that it may be the reason for async scenario. Or there are some other multi-threaded scenarios.
